src/common/transformations/src/transformations/common_optimizations/reshape_sequence_fusion.cpp

```diff
@@ -17,21 +17,53 @@
 NGRAPH_RTTI_DEFINITION(ngraph::pass::ReshapeSequenceFusion, "ReshapeSequenceFusion", 0);
 
 namespace {
+OPENVINO_SUPPRESS_DEPRECATED_START
 bool has_valid_pattern(const std::shared_ptr<ngraph::Node> & node) {
     auto const_node = std::dynamic_pointer_cast<ngraph::opset8::Constant>(node);
-    if (!const_node) return false;
+    if (!const_node) {
+        // Lower bound of the value
+        ngraph::HostTensorPtr lb = std::make_shared<ngraph::HostTensor>(node->output(0));
+        const bool is_lb_evaluated = node->evaluate_lower({lb});
```

Comment:
since we eventually can end up using both upper and lower bounds -- we can call `evaluate_both_bounds(node->output(0))`

Reply:
As we don't need upper bounds evaluation if all of the lower bounds are higher than 0, I used the `evaluate_lower_bound` and `evaluate_upper_bound` separately.